Var/src/var.c

```diff
@@ -0,0 +1,84 @@
+#include "var.h"
+#define PRINT_VAR(var_type, format) printf(""#format" (%p)\n", *(var_type*)ptr->valptr, ptr->valptr)
+
+#define OPTIMIZED_REALLOC(var_type, old_size, size) \
+	if (size > old_size) { \
+		/* When scaling up free/malloc is faster */ \
+		free_fn(ptr->valptr); \
+		if (!(valptr = malloc_fn(size))) { \
+			printf("Failed to allocate memory, cancelling"); \
+			return; \
+		} \
+	} \
+	else if (old_size > size){ \
+		/* When scaling down realloc is faster */ \
+		if (!(valptr = realloc(ptr->valptr, size))) { \
+			printf("Failed to reallocate memory, cancellinng\n"); \
+			return; \
+		} \
+	} \
+	else valptr = ptr->valptr; \
+
+#define ALLOC_VAL(var_type, enum_type, size)\
+	void *valptr; \
+	if (ptr->valptr){ /* valptr is not NULL, therefore a var already exist and holds some kind of value and type */ \
+		if (ptr->type == STRING) free_fn(*(char **)ptr->valptr); \
+		switch (ptr->type){ \
+			case INTEGER: OPTIMIZED_REALLOC(int, sizeof(int), size); break; \
```

Comment:
Please find a better way to do this. Maybe X-Macros?

Reply:
Not sure what you meant here. Isn't what i used an X macro?